### PR TITLE
[Doc] Initial search documentation

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Adding initial search documentation.

**🆕 Changelog**

- [Document tree](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/6549e1343d9c21baee54ab6defd0da74bc195ae0)
- [Document spatial search result](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/86b06ac3e60f2cced3693dff67b99f3e620dc8b7)
- [Document octree](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/206fa691554835ae91a19624c47393d3c485e867)
- [Document kd_tree](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/5022e9b7b827c2168a0044abfa58827c75449ad2)
- [Document bins static](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/cfe452ebd5c35e40e4682358425a8ff7373ebc82)
- [Document bins dynamic](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/f2b6d512900c604fb10003cabc3ac4b92ec42acc)
- [Document GeometricalObjectsBins](https://github.com/KratosMultiphysics/Kratos/pull/12461/commits/0a74100260e75462417e44e9aaf75c233fd7d470)

### CODE REVIEW

**Review by roigcarlo — DISMISSED** [review]

**Comment by sunethwarna** [comment]
@loumalouomega nice :) Btw, we don't put png and other stuff in the Kratos repo, we put them in https://github.com/KratosMultiphysics/Documentation.

**Review by sunethwarna — CHANGES_REQUESTED** [review]
@loumalouomega nice :) Btw, we don't put png and other stuff in the Kratos repo, we put them in https://github.com/KratosMultiphysics/Documentation.

I had to block since the auto merge was enabled.

**Comment by loumalouomega** [comment]
> @loumalouomega nice :) Btw, we don't put png and other stuff in the Kratos repo, we put them in [KratosMultiphysics/Documentation](https://github.com/KratosMultiphysics/Documentation).
> 
> I had to block since the auto merge was enabled.

Done

File: docs/pages/Kratos/Spatial_Containers/kd_tree.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+title: KDTree
+keywords: search spatial_container core
+tags: [search spatial_container]
+sidebar: kratos_core_search
+summary: A k-d tree (short for k-dimensional tree) is a space-partitioning data structure for organizing points in a k-dimensional space.
+---
+
+# KDTree
+
+![A 3-d tree](https://github.com/KratosMultiphysics/Documentation/blob/master/Wiki_files/Search/3dtree.png)
```
Comment (sunethwarna) [inline docs/pages/Kratos/Spatial_Containers/kd_tree.md]: Small question. I am not sure whther the images will be displayed or a link to images will be displayed. If it is the latter, then I would do it as the following:
https://github.com/KratosMultiphysics/Kratos/blob/e55171cd1820a583298ef16859bf6a9ca6bac685/docs/pages/Applications/Optimization_Application/Filtering/Explicit_Vertex_Morphing.md?plain=1#L13-L16

Comment (loumalouomega) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: A priori the images should work as I added them

**Comment by loumalouomega** [comment]
@roigcarlo do you knwo how to fix the doc build?

**Comment by roigcarlo** [comment]
Mm its complains about reaching the max depth on the nav menus, for that we need to change how the whole menu works, so would tell you to put everything one level above and create a new section in the top menu.

For what I can read, the search is big enough to have its own section in the top nav.

File: docs/pages/Kratos/Spatial_Containers/kd_tree.md (hunk 1 above)
Comment (roigcarlo) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: @sunethwarna I think vicente is right, once transpiled to html the markdown link/image format is automatically translated to a in `` tag

Comment (sunethwarna) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: good to know for future :)

**Comment by sunethwarna** [comment]
Max menu depth is 3 which is a limitation from the theme we are using. So as @roigcarlo said, you have put these one level up.

**Review by sunethwarna — DISMISSED** [review]

File: docs/pages/Kratos/Spatial_Containers/kd_tree.md (hunk 1 above)
Comment (sunethwarna) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: I dont think images are working like that, since the direct link does not give the raw data. :/

Comment (loumalouomega) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: Yes, you were right

Comment (roigcarlo) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: So did it make the conversion or not? who was right?

Comment (loumalouomega) [reply docs/pages/Kratos/Spatial_Containers/kd_tree.md]: @sunethwarna was right

**Comment by sunethwarna** [comment]
@loumalouomega It still does not show the images :/

**Comment by loumalouomega** [comment]
> @loumalouomega It still does not show the images :/

The menu appears empty

**Comment by loumalouomega** [comment]
..................

> @loumalouomega It still does not show the images :/

image

Yes, it does

**Comment by loumalouomega** [comment]
> ..................
> 
> > > @loumalouomega It still does not show the images :/
> > 
> > 
> > The menu appears empty
> 
> I refactor, let's see if works

It works, can we merge?

**Comment by loumalouomega** [comment]
Hello? @sunethwarna

**Comment by loumalouomega** [comment]
@roigcarlo

**Comment by roigcarlo** [comment]
To me looks good, but @sunethwarna has to lift the block

File: docs/pages/Kratos/Spatial_Containers/Trees_And_Searches/tree.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+title: Tree
+keywords: search spatial_container core
+tags: [search spatial_container]
+sidebar: kratos_core_search
+summary: A tree is a widely used abstract data type that represents a hierarchical tree structure with a set of connected nodes.
+---
+
+# Tree
+
+![A generic, and so non-binary, unsorted, and with duplicated labels; arbitrary diagram of a tree](https://github.com/KratosMultiphysics/Documentation/blob/master/Wiki_files/Search/tree.png)
```
Comment (sunethwarna) [inline docs/pages/Kratos/Spatial_Containers/Trees_And_Searches/tree.md]:
```suggestion
![A generic, and so non-binary, unsorted, and with duplicated labels; arbitrary diagram of a tree](https://github.com/KratosMultiphysics/Documentation/blob/master/Wiki_files/Search/tree.png?raw=true)
```

**Review by sunethwarna — COMMENTED** [review]
minor

**Comment by loumalouomega** [comment]
> minor

Done

**Review by sunethwarna — APPROVED** [review]
Thanks @loumalouomega

**Pull request merged**